File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [1]*len(nums)

        n= len(nums)
        i = 1

        while i < n:
            output[i] = output[i-1] * nums[i-1]
            i += 1

        i = n-2
        postfix = nums[n-1]
        while i >= 0 :
            output[i] = output[i] * postfix
            postfix = postfix * nums[i]
            i -=1

        return output 

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0 , r = nums.length-1;

        while(l < r){
            if ( nums[l] > nums[r]){
                if (nums[r-1] < nums[r]){
                    r--;
                }
                else{
                    return nums[r];
                }
            }
            else{
                return nums[l];
            }
        }
        return nums[r];
    }
}
